src/vm/arm64/asmhelpers.asm

```diff
@@ -405,6 +403,106 @@ Exit
         ret      lr
     LEAF_END
 
+; void JIT_UpdateWriteBarrierState(bool skipEphemeralCheck)
+;
+; Update shadow copies of the various state info required for barrier
+;
+; State info is contained in a literal pool at the end of the function
+; Placed in text section so that it is close enough to use ldr literal and still
+; be relocatable. Eliminates need for PREPARE_EXTERNAL_VAR in hot code.
+;
+; Align and group state info together so it fits in a single cache line
+; and each entry can be written atomically
+;
+    WRITE_BARRIER_ENTRY JIT_UpdateWriteBarrierState
+        PROLOG_SAVE_REG_PAIR   fp, lr, #-16!
+
+        ; x0-x7 will contain intended new state
+        ; x8 will preserve skipEphemeralCheck
+        ; x12 will be used for pointers
+
+        mov      x8, x0
+
+        adrp     x12, g_lowest_address
+        ldr      x0, [x12, g_lowest_address]
+
+        adrp     x12, g_highest_address
+        ldr      x1, [x12, g_highest_address]
+
+#ifdef WRITE_BARRIER_CHECK
+        adrp     x12, g_GCShadow
+        ldr      x2, [x12, g_GCShadow]
+#else
+        mov      x2, xzr
+#endif
+
+#ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+        adrp     x12, g_sw_ww_enabled_for_gc_heap
+        ldrb     w3, [x12, g_sw_ww_enabled_for_gc_heap]
+        cbz      x3, SOFTWARE_WRITE_WATCH_DISABLED
+        adrp     x12, g_sw_ww_table
+        ldr      x3, [x12, g_sw_ww_table]
+SOFTWARE_WRITE_WATCH_DISABLED
+#else
+        mov      x3, xzr
+#endif
+
+        adrp     x12, g_ephemeral_low
+        ldr      x4, [x12, g_ephemeral_low]
+
+        adrp     x12, g_ephemeral_high
+        ldr      x5, [x12, g_ephemeral_high]
+
+        cbz      x8, EphemeralCheckEnabled
+        movz     x4, #0
+        movn     x5, #0
+EphemeralCheckEnabled
+
+        adrp     x6, g_card_table
+        ldr      x6, [x12, g_card_table]
+
+#ifdef FEATURE_MANUALLY_MANAGED_CARD_BUNDLES
+        adrp     x7, g_card_bundle_table
+        ldr      x7, [x12, g_card_bundle_table]
+#else
+        mov      x7, xzr
+#endif
+
+        ; Update wbs state
+
+        adr  x12, wbs_begin
+        stp  x0, x1, [x12], 16
+        stp  x2, x3, [x12], 16
+        stp  x4, x5, [x12], 16
+        stp  x6, x7, [x12], 16
+
+        ; Force update to other processors
+        dmb      ishst
```

Comment:
@jkotas I am going to leave this in.  It is lighter weight than FlushProcessWriteBuffers and arm64 is still not GC stable.  Since this should not be called very often it probably will not affect performance.  I may revisit when arm64 is 100% GC stable.